core/src/main/scala/org/apache/spark/mllib/evaluation/RichBinaryClassificationMetrics.scala

```diff
@@ -38,16 +40,107 @@ import org.apache.spark.rdd.RDD
  *                partition boundaries.
  */
 class RichBinaryClassificationMetrics(
-  override val scoreAndLabels: RDD[(Double, Double)],
-  override val numBins: Int
-) extends BinaryClassificationMetrics(scoreAndLabels, numBins) {
+  val scoreAndLabels: RDD[(Double, Double)],
+  val numBins: Int
+) {
```

Comment:
why the base class was dropped?

Reply:
Wondering this too. We should be able to re-use some thing, no, @Jauntbox ?

Reply:
We used to extend Spark's class since we were only partially overriding it, but I don't think we need to extend that class anymore. We're completely copying it here so aren't actually sharing code with it anymore.

Is there anything to gain by extending it?

Reply:
If you arent extending it, then we should move `RichBinaryClassificationMetrics` out of that package and consider renaming it to `BinaryClassificationMetrics`

Reply:
(this might be a breaking change though)

Reply:
Yeah, renaming it makes sense. I can check if moving it causes a lot of trouble.

Reply:
I renamed to `SparkBinaryClassifiationMetrics` to make it clearer. It has to live in org.apache.spark.mllib.evaluation for its imports.

@nicodv Unfortunately we need to replicate everything since all the other metrics are built from `cumulativeCounts` and `confusions`.